spec/features/admin/index_spec.rb

```diff
@@ -93,14 +116,57 @@
-        expect(page).to have_content("Name: #{@customer_2.first_name} #{@customer_2.last_name} | Number of Successful Transactions: #{@customer_2.transaction_ct("success")}")
-        expect(page).to have_content("Name: #{@customer_3.first_name} #{@customer_3.last_name} | Number of Successful Transactions: #{@customer_3.transaction_ct("success")}")
-        expect(page).to have_content("Name: #{@customer_5.first_name} #{@customer_5.last_name} | Number of Successful Transactions: #{@customer_5.transaction_ct("success")}")
-        expect(page).to have_content("Name: #{@customer_7.first_name} #{@customer_7.last_name} | Number of Successful Transactions: #{@customer_7.transaction_ct("success")}")
+      visit "/admin/dashboard"
+
+       within("#dashboard-customers") do
+        expect(page).to have_content("Name: #{@customer_1.first_name} #{@customer_1.last_name} | Successful Transactions: #{@customer_1.transaction_ct("success")}")
+        expect(page).to have_content("Name: #{@customer_2.first_name} #{@customer_2.last_name} | Successful Transactions: #{@customer_2.transaction_ct("success")}")
+        expect(page).to have_content("Name: #{@customer_3.first_name} #{@customer_3.last_name} | Successful Transactions: #{@customer_3.transaction_ct("success")}")
+        expect(page).to have_content("Name: #{@customer_5.first_name} #{@customer_5.last_name} | Successful Transactions: #{@customer_5.transaction_ct("success")}")
+        expect(page).to have_content("Name: #{@customer_7.first_name} #{@customer_7.last_name} | Successful Transactions: #{@customer_7.transaction_ct("success")}")
+      end 
+  end
+
+  it "has a section for 'Incomplete Invoices' that contains a list of the unique id's of all invoices that have items that have 
+    not yet been shipped" do
+      visit "/admin/dashboard"
+
+      within("#dashboard-incomplete_invoices") do
+        expect(page).to have_content(@invoice_1.id, count: 1)
+        expect(page).to have_content(@invoice_2.id, count: 1)
+        expect(page).to have_no_content(@invoice_3.id)
+      end 
+  end
+
+  it "each invoice id links to that admin invoice's show page" do
+
+      visit "/admin/dashboard"
+      within("#dashboard-incomplete_invoices") do
+        expect(page).to have_link("#{@invoice_1.id}")
+      end 
+      click_link "#{@invoice_1.id}"
+      expect(current_path).to eq(admin_invoice_path("#{@invoice_1.id}"))
+
+      visit "/admin/dashboard"
+      within("#dashboard-incomplete_invoices") do
+        expect(page).to have_link("#{@invoice_2.id}")
+      end 
+      click_link "#{@invoice_2.id}"
+      expect(current_path).to eq(admin_invoice_path("#{@invoice_2.id}"))
+  end
+
+  it "next to each invoice is the date created formatted like 'Monday, July 18, 2019
+    and the invoices are listed in order from oldest to newest" do
+    
+    visit "/admin/dashboard"
+
+      within("#dashboard-incomplete_invoices") do
+        expect("#{@invoice_2.id}").to appear_before("#{@invoice_1.id}")
```

Comment:
I don't believe this test is covering the correct order indicated in the User Story (Oldest to Newest). `@invoice_2` appears after `@invoice_1` in the `before :each` block; Meaning `@invoice_1` would be the oldest and accordingly should appear first.

This was not apparent looking at the `save_and_open_page` result as only the date is shown and not the time. But when I booted up the rails server with the csv data, I was able to confirm the page had the newest invoices first.

Reply:
I will change the test to reflect the removal of .reverse_order in the view and push up the changes.